src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala

```diff
@@ -0,0 +1,331 @@
+  }
+
+  private def resolvingBranch(from: BigInt)(state: ImporterState): Receive =
+    running(ImporterState.resolvingBranch(from, state))
+
+  private def start(): Unit = {
+    log.debug("Starting Regular Sync, current best block is {}", startingBlockNumber)
+    fetcher ! BlockFetcher.Start(self, startingBlockNumber)
+    context become running(ImporterState.initial)
+  }
+
+  private def pickBlocks(state: ImporterState): Unit = {
+    val msg = state.resolvingBranchFrom.fold[BlockFetcher.FetchMsg](
+      BlockFetcher.PickBlocks(syncConfig.blocksBatchSize)
+    )(
+      from => BlockFetcher.StrictPickBlocks(from, startingBlockNumber)
+    )
+
+    fetcher ! msg
+  }
+
+  private def importBlocks(blocks: NonEmptyList[Block]): ImportFn =
+    importWith {
+      log.debug("Attempting to import blocks starting from {}", Block.number(blocks.head))
+      Future
+        .successful(resolveBranch(blocks))
+        .flatMap {
+          case Right(blocksToImport) => handleBlocksImport(blocksToImport)
+          case Left(resolvingFrom) => Future.successful(ResolvingBranch(resolvingFrom))
+        }
+    }
+
+  private def handleBlocksImport(blocks: List[Block]): Future[NewBehavior] =
+    tryImportBlocks(blocks)
+      .map { value =>
+        val (importedBlocks, errorOpt) = value
+        log.info("Imported blocks {}", importedBlocks.map(Block.number).mkString(","))
+
+        errorOpt match {
+          case None => Running
+          case Some(err) =>
+            log.error("Block import error {}", err)
+            val notImportedBlocks = blocks.drop(importedBlocks.size)
+
+            err match {
+              case e: MissingNodeException =>
+                fetcher ! BlockFetcher.FetchStateNode(e.hash)
+                ResolvingMissingNode(NonEmptyList(notImportedBlocks.head, notImportedBlocks.tail))
+              case _ =>
+                val invalidBlockNr = Block.number(notImportedBlocks.head)
+                fetcher ! BlockFetcher.InvalidateBlocksFrom(invalidBlockNr, err.toString)
+                Running
+            }
+        }
+      }
+
+  private def tryImportBlocks(blocks: List[Block], importedBlocks: List[Block] = Nil)(
+      implicit ec: ExecutionContext): Future[(List[Block], Option[Any])] =
+    if (blocks.isEmpty) {
+      Future.successful((importedBlocks, None))
+    } else {
+      val restOfBlocks = blocks.tail
+      ledger
+        .importBlock(blocks.head)
+        .flatMap {
+          case BlockImportedToTop(_) =>
+            tryImportBlocks(restOfBlocks, blocks.head :: importedBlocks)
+
+          case ChainReorganised(_, newBranch, _) =>
+            tryImportBlocks(restOfBlocks, newBranch.reverse ::: importedBlocks)
+
+          case DuplicateBlock | BlockEnqueued =>
+            tryImportBlocks(restOfBlocks, importedBlocks)
+
+          case err @ (UnknownParent | BlockImportFailed(_)) =>
+            log.error(s"block ${Block.number(blocks.head)} import failed")
```

Comment:
should start with uppercase and probably you could use `{}` like in the rest of the logs here